npm-publish/action.yml

```diff
@@ -0,0 +1,41 @@
+---
+
+name: Publish package to npm
+description: This action automates various aspects of publishing a package to the npm registry including running tests, versioning, and tagging.
+inputs:
+  release_type:
+    description: 'The npm version type we are releasing.'
+    required: true
+    type: choice
+    default: 'patch'
+    options:
+      - patch
+      - minor
+      - major
+  NPM_TOKEN:
+    description: 'The npm token used to publish the package.'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout Code
+      uses: actions/checkout@v3
+
+    - name: Setup Node
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16
```

Comment:
I would make this configurable. Say,

```yaml
inputs:
  ...
  node-version:
    description: Node.js version to use
    required: false
    default: 'lts/*'
```

and then:

```yaml
    - name: Setup Node
      uses: actions/setup-node@v3
      with:
        node-version: ${{ inputs.node-version }}
```

This approach has some advantages:
* we don't have to update `node-version` when a new Node.js gets released - by default, we will use the current LTS release;
* some of our projects require outdated versions of Node.js; by providing a configurable option, we allow them to use this action.

Reply:
Good idea; done in 8b4e19f